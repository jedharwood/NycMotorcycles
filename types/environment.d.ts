declare global {
  namespace NodeJS {
    interface ProcessEnv {
      NEXT_PUBLIC_API_MOCKING: string;
      SELECTOR_PRODUCT: string;
      SELECTOR_PRODUCT_PRICE: string;
      SELECTOR_PRODUCT_TIME: string;
      SELECTOR_PRODUCT_TITLE: string;
      SELECTOR_PRODUCT_BID: string;
      YAHOO_AUCTION_PROFILE_PAGE_URL: string;
      MAILGUN_API_KEY: string;
      MAILGUN_DOMAIN: string;
      DELIVERY_EMAIL: string;
    }
  }
}

export {};
