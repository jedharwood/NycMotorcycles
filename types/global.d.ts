type TimeRemaining = {
  time: number | undefined;
  unit: 'days' | 'hours' | 'minutes';
};

type ActiveAuction = {
  title?: string;
  url?: string;
  image: GridImage;
  bidders?: string;
  timeRemaining?: TimeRemaining;
  currentPrice?: string;
  promptDecisionPrice?: string;
};

type GridImage = {
  imageSrc: string;
  imageAlt: string;
  width?: number;
  height?: number;
  onImageClick?: () => void;
};

type ArchiveBikeKey = keyof typeof archive;

type ContactFormData = {
  email: string;
  senderName: string;
  subject: string;
  message: string;
};

type InputFieldProps = {
  type: 'email' | 'text' | 'text-area';
  placeholder: string;
  name: 'email' | 'senderName' | 'subject' | 'message';
  register: UseFormRegister<ContactFormData>;
  error: FieldError | undefined;
  label: string;
};

type ComponentColour = 'green' | 'red';

type SoldOrCall = 'sold' | 'call';

type TheNumbersOneToFour = 1 | 2 | 3 | 4;

type NavLinkDetails = {
  href: string;
  text: string;
};

type FooterProps = {
  address: string;
  disclaimer: string;
  chevronText: string;
  altTextInstagramButton: string;
};

type ChildElementPosition = 'under-title' | 'bottom';

type TextDisplayProps = {
  title?: string;
  textContent?: string[];
  childElement?: JSX.Element;
  childElementPosition?: ChildElementPosition;
  isOpaque?: boolean;
  hasBorder?: boolean;
  textContentCentred?: boolean;
  borderColour?: ComponentColour;
  subTitle?: string;
  footer?: string;
};