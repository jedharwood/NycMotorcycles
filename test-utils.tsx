import React, { ReactElement } from 'react';
import { render, RenderOptions } from '@testing-library/react';
import { IntlProvider } from 'react-intl';
import en from './languages/en.json';
import { MemoryRouterProvider } from 'next-router-mock/MemoryRouterProvider';
import { QueryClient, QueryClientProvider } from 'react-query';

const AllTheProviders = ({ children }: { children: React.ReactNode }) => {
  const queryClient = new QueryClient();

  return (
    <MemoryRouterProvider>
      <QueryClientProvider client={queryClient}>
        <IntlProvider locale='en' messages={en}>
          {children}
        </IntlProvider>
      </QueryClientProvider>
    </MemoryRouterProvider>
  );
};

const renderWithProviders = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'wrapper'>
) => render(ui, { wrapper: AllTheProviders, ...options });

export * from '@testing-library/react';
export { renderWithProviders as render };
