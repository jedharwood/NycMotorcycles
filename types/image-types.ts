import { StaticImageData } from 'next/image';

export type StaticImage = {
  image: StaticImageData;
  altText: string;
  onImageClick?: () => void;
};

export type ImageIndex = {
  [key: string]: StaticImage;
};

export const blankStaticImage: StaticImage = {
  image: {
    src: '',
    height: 0,
    width: 0,
  },
  altText: '',
};