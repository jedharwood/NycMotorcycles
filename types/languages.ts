import en from '../languages/en.json';
import ja from '../languages/ja.json';

export const langs = {
    en: 'en',
    ja: 'ja',
};

type Messages = {
    [key: string]: {
        [key: string]: string;
    };
}; 

export const messages: Messages = {
    en,
    ja,
};