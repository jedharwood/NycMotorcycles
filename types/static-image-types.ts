import { StaticImageData } from 'next/image'

export type StaticImage = {
    image: StaticImageData;
    altText: string;
    onImageClick?: () => void;
}

export type StaticImageCatalog = {
    [key: string]: StaticImage;
  };