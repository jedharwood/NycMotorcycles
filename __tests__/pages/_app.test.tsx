import userEvent from '@testing-library/user-event';
import { AppProps } from 'next/app';
import mockRouter from 'next-router-mock';

import { render, act, screen, waitFor } from '@/test-utils';
import { langs } from '@/types/languages';

import App from '../../pages/_app';

jest.mock('next/router', () => jest.requireActual('next-router-mock'));

const renderApp = (Component: React.ComponentType, pageProps: any = {}) => {
    const props: AppProps = {
        Component,
        pageProps,
        router: mockRouter as any,
    };

    return render(<App {...props} />);
};

const activeNavLinkClasses: string = 'underline opacity-100';
const navLinkRacingDesktopTestId: string = 'nav-link-racing-desktop';

const checkHomePageNavLinkIsActive = async (): Promise<void> => {
    await waitFor(() => {
        const homeNavLink: HTMLElement = screen.getByTestId('nav-link-home-desktop');
        expect(homeNavLink).toHaveClass(activeNavLinkClasses);
    });

    await checkRacingPageNavLinkIsActive(false);
};

const checkRacingPageNavLinkIsActive = async (
    shouldBeActive: boolean = true
): Promise<void> => {
    await waitFor(() => {
        const racingNavLink: HTMLElement = screen.getByTestId(navLinkRacingDesktopTestId);
        shouldBeActive
            ? expect(racingNavLink).toHaveClass(activeNavLinkClasses)
            : expect(racingNavLink).not.toHaveClass(activeNavLinkClasses);
    });
};

describe('App', () => {
    beforeEach(() => {
        mockRouter.setCurrentUrl('/');
        mockRouter.locale = langs.en;
    });

    afterEach(() => {
        jest.clearAllMocks();
    });

    const TestPage = (): JSX.Element => <div>Test Page</div>;

    it('renders Layout in English and matches snapshot', () => {
        const { container } = renderApp(TestPage);

        expect(container).toMatchSnapshot();
    });

    it('renders Layout in Japanese and matches snapshot', () => {
        mockRouter.locale = langs.ja;
        const { container } = renderApp(TestPage);

        expect(container).toMatchSnapshot();
    });

    describe('Header component', () => {
        it('should set active style on clicked nav link', async () => {
            renderApp(TestPage);

            await act(() => {
                const racingNavLink: HTMLElement = screen.getByTestId(
                    navLinkRacingDesktopTestId
                );
                userEvent.click(racingNavLink);
            });

            await checkRacingPageNavLinkIsActive();
        });

        it('should set active style on home link when logo image is clicked', async () => {
            mockRouter.setCurrentUrl('/racing');
            renderApp(TestPage);

            await checkRacingPageNavLinkIsActive();

            await act(() => {
                const nycmcLogo: HTMLElement = screen.getByAltText(
                    'New york city motorcycles logo'
                );
                userEvent.click(nycmcLogo);
            });

            await checkHomePageNavLinkIsActive();
        });

        it('should set active style on home link when header h1 page name is clicked', async () => {
            mockRouter.setCurrentUrl('/racing');
            renderApp(TestPage);

            await checkRacingPageNavLinkIsActive();

            await act(() => {
                const homepageLinks = screen.getAllByRole('link', {
                    name: /new york city motorcycles/i,
                });
                const textLink = homepageLinks.find((link) => link.querySelector('h1'));
                userEvent.click(textLink!);
            });

            await checkHomePageNavLinkIsActive();
        });

        // can maybe click test the flag lang switcher...
        // test click mibile menu
    });
});
