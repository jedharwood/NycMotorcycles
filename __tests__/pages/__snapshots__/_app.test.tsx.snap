// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders Layout in English and matches snapshot 1`] = `
<div>
  <div
    class="w-full bg-hero bg-cover bg-fixed bg-center bg-no-repeat md:px-6"
    id="layout-outer"
  >
    <div
      class="mx-auto max-w-6xl bg-stone-600 bg-opacity-60 py-6 sm:p-6"
      id="layout-inner"
    >
      <header
        class="text-stone-50"
      >
        <div
          class="mr-6 flex justify-end sm:mr-0"
        >
          <button>
            <svg
              class="rounded-sm h-6 w-9 shadow-lg card-hover"
              viewBox="0 0 513 342"
              xmlns="http://www.w3.org/2000/svg"
            >
              <title>
                English
              </title>
              <path
                d="M0 0h513v342H0z"
                fill="#FFF"
              />
              <path
                d="M0 0h513v26.3H0zm0 52.6h513v26.3H0zm0 52.6h513v26.3H0zm0 52.6h513v26.3H0zm0 52.7h513v26.3H0zm0 52.6h513v26.3H0zm0 52.6h513V342H0z"
                fill="#D80027"
              />
              <path
                d="M0 0h256.5v184.1H0z"
                fill="#2E52B2"
              />
              <path
                d="m47.8 138.9-4-12.8-4.4 12.8H26.2l10.7 7.7-4 12.8 10.9-7.9 10.6 7.9-4.1-12.8 10.9-7.7zm56.3 0-4.1-12.8-4.2 12.8H82.6l10.7 7.7-4 12.8 10.7-7.9 10.8 7.9-4-12.8 10.7-7.7zm56.5 0-4.3-12.8-4 12.8h-13.5l11 7.7-4.2 12.8 10.7-7.9 11 7.9-4.2-12.8 10.7-7.7zm56.2 0-4-12.8-4.2 12.8h-13.3l10.8 7.7-4 12.8 10.7-7.9 10.8 7.9-4.3-12.8 11-7.7zM100 75.3l-4.2 12.8H82.6L93.3 96l-4 12.6 10.7-7.8 10.8 7.8-4-12.6 10.7-7.9h-13.4zm-56.2 0-4.4 12.8H26.2L36.9 96l-4 12.6 10.9-7.8 10.6 7.8L50.3 96l10.9-7.9H47.8zm112.5 0-4 12.8h-13.5l11 7.9-4.2 12.6 10.7-7.8 11 7.8-4.2-12.6 10.7-7.9h-13.2zm56.5 0-4.2 12.8h-13.3l10.8 7.9-4 12.6 10.7-7.8 10.8 7.8-4.3-12.6 11-7.9h-13.5zm-169-50.6-4.4 12.6H26.2l10.7 7.9-4 12.7L43.8 50l10.6 7.9-4.1-12.7 10.9-7.9H47.8zm56.2 0-4.2 12.6H82.6l10.7 7.9-4 12.7L100 50l10.8 7.9-4-12.7 10.7-7.9h-13.4zm56.3 0-4 12.6h-13.5l11 7.9-4.2 12.7 10.7-7.9 11 7.9-4.2-12.7 10.7-7.9h-13.2zm56.5 0-4.2 12.6h-13.3l10.8 7.9-4 12.7 10.7-7.9 10.8 7.9-4.3-12.7 11-7.9h-13.5z"
                fill="#FFF"
              />
            </svg>
          </button>
        </div>
        <div
          class="flex justify-center pb-6"
        >
          <a
            href="/"
          >
            <img
              alt="New york city motorcycles logo"
              src="[object Object]"
            />
          </a>
        </div>
        <div
          class="flex justify-center"
        >
          <a
            href="/"
          >
            <h1
              class="text-lg tracking-widest md:text-xl"
            >
              NEW YORK CITY MOTORCYCLES
            </h1>
          </a>
        </div>
        <nav
          class="mt-2 mb-6 rounded-md bg-stone-600 bg-opacity-90 py-2 px-6 shadow-lg"
        >
          <div
            class="container mx-auto flex flex-wrap justify-center"
          >
            <button
              aria-controls="navbar-default"
              aria-expanded="false"
              class="inline-flex items-center rounded-lg opacity-80 hover:opacity-100 focus:outline-none focus:ring-2 focus:ring-stone-50 md:hidden"
              type="button"
            >
              <span
                class="sr-only"
              >
                Open main menu
              </span>
              <img
                alt="Burger menu icon"
                src="[object Object]"
              />
            </button>
            <div
              class="hidden w-full md:block md:w-auto"
            >
              <ul
                class="flex flex-wrap justify-center md:space-x-6 md:font-medium"
              >
                <li
                  class="flex justify-center uppercase"
                >
                  <a
                    class="block md:hover:underline md:hover:opacity-100 text-nowrap underline opacity-100"
                    href="/"
                  >
                    Home
                  </a>
                </li>
                <li
                  class="flex justify-center uppercase"
                >
                  <a
                    class="block md:hover:underline md:hover:opacity-100 text-nowrap opacity-80"
                    href="/active-auctions"
                  >
                    Active Auctions
                  </a>
                </li>
                <li
                  class="flex justify-center uppercase"
                >
                  <a
                    class="block md:hover:underline md:hover:opacity-100 text-nowrap opacity-80"
                    href="/sold-archive"
                  >
                    Sold Archive
                  </a>
                </li>
                <li
                  class="flex justify-center uppercase"
                >
                  <a
                    class="block md:hover:underline md:hover:opacity-100 text-nowrap opacity-80"
                    href="/history"
                  >
                    History
                  </a>
                </li>
                <li
                  class="flex justify-center uppercase"
                >
                  <a
                    class="block md:hover:underline md:hover:opacity-100 text-nowrap opacity-80"
                    href="/racing"
                  >
                    Racing
                  </a>
                </li>
                <li
                  class="flex justify-center uppercase"
                >
                  <a
                    class="block md:hover:underline md:hover:opacity-100 text-nowrap opacity-80"
                    href="/contact"
                  >
                    Contact
                  </a>
                </li>
                <li
                  class="flex justify-center uppercase"
                >
                  <a
                    class="block md:hover:underline md:hover:opacity-100 text-nowrap opacity-80"
                    href="/consign"
                  >
                    Consign
                  </a>
                </li>
                <li
                  class="flex justify-center uppercase"
                >
                  <a
                    class="block md:hover:underline md:hover:opacity-100 text-nowrap opacity-80"
                    href="/invest"
                  >
                    Invest
                  </a>
                </li>
                <li
                  class="flex justify-center uppercase"
                >
                  <a
                    class="block md:hover:underline md:hover:opacity-100 text-nowrap opacity-80"
                    href="/download"
                  >
                    Download
                  </a>
                </li>
              </ul>
            </div>
          </div>
          <div
            class="invisible h-0 transition[height] overflow-hidden duration-300 ease-in-out md:hidden"
          >
            <ul
              class="mt-4 mb-2 grid space-y-2"
            >
              <li
                class="flex justify-center uppercase"
              >
                <a
                  class="block md:hover:underline md:hover:opacity-100 text-nowrap underline opacity-100"
                  href="/"
                >
                  Home
                </a>
              </li>
              <li
                class="flex justify-center uppercase"
              >
                <a
                  class="block md:hover:underline md:hover:opacity-100 text-nowrap opacity-80"
                  href="/active-auctions"
                >
                  Active Auctions
                </a>
              </li>
              <li
                class="flex justify-center uppercase"
              >
                <a
                  class="block md:hover:underline md:hover:opacity-100 text-nowrap opacity-80"
                  href="/sold-archive"
                >
                  Sold Archive
                </a>
              </li>
              <li
                class="flex justify-center uppercase"
              >
                <a
                  class="block md:hover:underline md:hover:opacity-100 text-nowrap opacity-80"
                  href="/history"
                >
                  History
                </a>
              </li>
              <li
                class="flex justify-center uppercase"
              >
                <a
                  class="block md:hover:underline md:hover:opacity-100 text-nowrap opacity-80"
                  href="/racing"
                >
                  Racing
                </a>
              </li>
              <li
                class="flex justify-center uppercase"
              >
                <a
                  class="block md:hover:underline md:hover:opacity-100 text-nowrap opacity-80"
                  href="/contact"
                >
                  Contact
                </a>
              </li>
              <li
                class="flex justify-center uppercase"
              >
                <a
                  class="block md:hover:underline md:hover:opacity-100 text-nowrap opacity-80"
                  href="/consign"
                >
                  Consign
                </a>
              </li>
              <li
                class="flex justify-center uppercase"
              >
                <a
                  class="block md:hover:underline md:hover:opacity-100 text-nowrap opacity-80"
                  href="/invest"
                >
                  Invest
                </a>
              </li>
              <li
                class="flex justify-center uppercase"
              >
                <a
                  class="block md:hover:underline md:hover:opacity-100 text-nowrap opacity-80"
                  href="/download"
                >
                  Download
                </a>
              </li>
            </ul>
          </div>
        </nav>
      </header>
      <div>
        Test Page
      </div>
      <footer
        class="mt-6 grid space-y-4 font-light tracking-widest text-stone-50"
      >
        <div
          class="flex justify-center"
        >
          <button
            class="opacity-80 hover:underline hover:opacity-100"
            data-testid="scroll-to-top-button"
            type="button"
          >
            <div
              class="flex justify-center"
            >
              <img
                alt="Scroll to top"
                src="[object Object]"
              />
            </div>
            Scroll to top
          </button>
        </div>
        <div
          class="flex justify-center"
        >
          <a
            class="opacity-80 hover:opacity-100"
            data-testid="instagram-button-footer"
            href="https://www.instagram.com/newyorkcitymotorcycles/"
            rel="noopener noreferrer"
            target="_blank"
          >
            <img
              alt="Instagram logo"
              src="[object Object]"
            />
          </a>
        </div>
        <h3
          class="md:text-md flex justify-center text-sm"
        >
          SHONAN COAST, JAPAN | LOS ANGELES
        </h3>
        <h4
          class="flex justify-center px-1 text-xs md:text-sm"
        >
          Vintage Motorcycles, Inc | All Rights Reserved
        </h4>
      </footer>
    </div>
  </div>
</div>
`;

exports[`App renders Layout in Japanese and matches snapshot 1`] = `
<div>
  <div
    class="w-full bg-hero bg-cover bg-fixed bg-center bg-no-repeat md:px-6"
    id="layout-outer"
  >
    <div
      class="mx-auto max-w-6xl bg-stone-600 bg-opacity-60 py-6 sm:p-6"
      id="layout-inner"
    >
      <header
        class="text-stone-50"
      >
        <div
          class="mr-6 flex justify-end sm:mr-0"
        >
          <button>
            <svg
              class="rounded-sm h-6 w-9 shadow-lg card-hover"
              viewBox="0 0 513 342"
              xmlns="http://www.w3.org/2000/svg"
            >
              <title>
                English
              </title>
              <path
                d="M0 0h513v342H0z"
                fill="#FFF"
              />
              <path
                d="M0 0h513v26.3H0zm0 52.6h513v26.3H0zm0 52.6h513v26.3H0zm0 52.6h513v26.3H0zm0 52.7h513v26.3H0zm0 52.6h513v26.3H0zm0 52.6h513V342H0z"
                fill="#D80027"
              />
              <path
                d="M0 0h256.5v184.1H0z"
                fill="#2E52B2"
              />
              <path
                d="m47.8 138.9-4-12.8-4.4 12.8H26.2l10.7 7.7-4 12.8 10.9-7.9 10.6 7.9-4.1-12.8 10.9-7.7zm56.3 0-4.1-12.8-4.2 12.8H82.6l10.7 7.7-4 12.8 10.7-7.9 10.8 7.9-4-12.8 10.7-7.7zm56.5 0-4.3-12.8-4 12.8h-13.5l11 7.7-4.2 12.8 10.7-7.9 11 7.9-4.2-12.8 10.7-7.7zm56.2 0-4-12.8-4.2 12.8h-13.3l10.8 7.7-4 12.8 10.7-7.9 10.8 7.9-4.3-12.8 11-7.7zM100 75.3l-4.2 12.8H82.6L93.3 96l-4 12.6 10.7-7.8 10.8 7.8-4-12.6 10.7-7.9h-13.4zm-56.2 0-4.4 12.8H26.2L36.9 96l-4 12.6 10.9-7.8 10.6 7.8L50.3 96l10.9-7.9H47.8zm112.5 0-4 12.8h-13.5l11 7.9-4.2 12.6 10.7-7.8 11 7.8-4.2-12.6 10.7-7.9h-13.2zm56.5 0-4.2 12.8h-13.3l10.8 7.9-4 12.6 10.7-7.8 10.8 7.8-4.3-12.6 11-7.9h-13.5zm-169-50.6-4.4 12.6H26.2l10.7 7.9-4 12.7L43.8 50l10.6 7.9-4.1-12.7 10.9-7.9H47.8zm56.2 0-4.2 12.6H82.6l10.7 7.9-4 12.7L100 50l10.8 7.9-4-12.7 10.7-7.9h-13.4zm56.3 0-4 12.6h-13.5l11 7.9-4.2 12.7 10.7-7.9 11 7.9-4.2-12.7 10.7-7.9h-13.2zm56.5 0-4.2 12.6h-13.3l10.8 7.9-4 12.7 10.7-7.9 10.8 7.9-4.3-12.7 11-7.9h-13.5z"
                fill="#FFF"
              />
            </svg>
          </button>
        </div>
        <div
          class="flex justify-center pb-6"
        >
          <a
            href="/"
          >
            <img
              alt="placeholder"
              src="[object Object]"
            />
          </a>
        </div>
        <div
          class="flex justify-center"
        >
          <a
            href="/"
          >
            <h1
              class="text-lg tracking-widest md:text-xl"
            >
              NEW YORK CITY MOTORCYCLES
            </h1>
          </a>
        </div>
        <nav
          class="mt-2 mb-6 rounded-md bg-stone-600 bg-opacity-90 py-2 px-6 shadow-lg"
        >
          <div
            class="container mx-auto flex flex-wrap justify-center"
          >
            <button
              aria-controls="navbar-default"
              aria-expanded="false"
              class="inline-flex items-center rounded-lg opacity-80 hover:opacity-100 focus:outline-none focus:ring-2 focus:ring-stone-50 md:hidden"
              type="button"
            >
              <span
                class="sr-only"
              >
                placeholder
              </span>
              <img
                alt="placeholder"
                src="[object Object]"
              />
            </button>
            <div
              class="hidden w-full md:block md:w-auto"
            >
              <ul
                class="flex flex-wrap justify-center md:space-x-6 md:font-medium"
              >
                <li
                  class="flex justify-center uppercase"
                >
                  <a
                    class="block md:hover:underline md:hover:opacity-100 text-nowrap opacity-80"
                    href="/"
                  >
                    placeholder
                  </a>
                </li>
                <li
                  class="flex justify-center uppercase"
                >
                  <a
                    class="block md:hover:underline md:hover:opacity-100 text-nowrap opacity-80"
                    href="/active-auctions"
                  >
                    placeholder
                  </a>
                </li>
                <li
                  class="flex justify-center uppercase"
                >
                  <a
                    class="block md:hover:underline md:hover:opacity-100 text-nowrap opacity-80"
                    href="/sold-archive"
                  >
                    placeholder
                  </a>
                </li>
                <li
                  class="flex justify-center uppercase"
                >
                  <a
                    class="block md:hover:underline md:hover:opacity-100 text-nowrap opacity-80"
                    href="/history"
                  >
                    placeholder
                  </a>
                </li>
                <li
                  class="flex justify-center uppercase"
                >
                  <a
                    class="block md:hover:underline md:hover:opacity-100 text-nowrap opacity-80"
                    href="/racing"
                  >
                    placeholder
                  </a>
                </li>
                <li
                  class="flex justify-center uppercase"
                >
                  <a
                    class="block md:hover:underline md:hover:opacity-100 text-nowrap opacity-80"
                    href="/contact"
                  >
                    placeholder
                  </a>
                </li>
                <li
                  class="flex justify-center uppercase"
                >
                  <a
                    class="block md:hover:underline md:hover:opacity-100 text-nowrap opacity-80"
                    href="/consign"
                  >
                    placeholder
                  </a>
                </li>
                <li
                  class="flex justify-center uppercase"
                >
                  <a
                    class="block md:hover:underline md:hover:opacity-100 text-nowrap opacity-80"
                    href="/invest"
                  >
                    placeholder
                  </a>
                </li>
                <li
                  class="flex justify-center uppercase"
                >
                  <a
                    class="block md:hover:underline md:hover:opacity-100 text-nowrap opacity-80"
                    href="/download"
                  >
                    placeholder
                  </a>
                </li>
              </ul>
            </div>
          </div>
          <div
            class="invisible h-0 transition[height] overflow-hidden duration-300 ease-in-out md:hidden"
          >
            <ul
              class="mt-4 mb-2 grid space-y-2"
            >
              <li
                class="flex justify-center uppercase"
              >
                <a
                  class="block md:hover:underline md:hover:opacity-100 text-nowrap opacity-80"
                  href="/"
                >
                  placeholder
                </a>
              </li>
              <li
                class="flex justify-center uppercase"
              >
                <a
                  class="block md:hover:underline md:hover:opacity-100 text-nowrap opacity-80"
                  href="/active-auctions"
                >
                  placeholder
                </a>
              </li>
              <li
                class="flex justify-center uppercase"
              >
                <a
                  class="block md:hover:underline md:hover:opacity-100 text-nowrap opacity-80"
                  href="/sold-archive"
                >
                  placeholder
                </a>
              </li>
              <li
                class="flex justify-center uppercase"
              >
                <a
                  class="block md:hover:underline md:hover:opacity-100 text-nowrap opacity-80"
                  href="/history"
                >
                  placeholder
                </a>
              </li>
              <li
                class="flex justify-center uppercase"
              >
                <a
                  class="block md:hover:underline md:hover:opacity-100 text-nowrap opacity-80"
                  href="/racing"
                >
                  placeholder
                </a>
              </li>
              <li
                class="flex justify-center uppercase"
              >
                <a
                  class="block md:hover:underline md:hover:opacity-100 text-nowrap opacity-80"
                  href="/contact"
                >
                  placeholder
                </a>
              </li>
              <li
                class="flex justify-center uppercase"
              >
                <a
                  class="block md:hover:underline md:hover:opacity-100 text-nowrap opacity-80"
                  href="/consign"
                >
                  placeholder
                </a>
              </li>
              <li
                class="flex justify-center uppercase"
              >
                <a
                  class="block md:hover:underline md:hover:opacity-100 text-nowrap opacity-80"
                  href="/invest"
                >
                  placeholder
                </a>
              </li>
              <li
                class="flex justify-center uppercase"
              >
                <a
                  class="block md:hover:underline md:hover:opacity-100 text-nowrap opacity-80"
                  href="/download"
                >
                  placeholder
                </a>
              </li>
            </ul>
          </div>
        </nav>
      </header>
      <div>
        Test Page
      </div>
      <footer
        class="mt-6 grid space-y-4 font-light tracking-widest text-stone-50"
      >
        <div
          class="flex justify-center"
        >
          <button
            class="opacity-80 hover:underline hover:opacity-100"
            data-testid="scroll-to-top-button"
            type="button"
          >
            <div
              class="flex justify-center"
            >
              <img
                alt="placeholder"
                src="[object Object]"
              />
            </div>
            placeholder
          </button>
        </div>
        <div
          class="flex justify-center"
        >
          <a
            class="opacity-80 hover:opacity-100"
            data-testid="instagram-button-footer"
            href="https://www.instagram.com/newyorkcitymotorcycles/"
            rel="noopener noreferrer"
            target="_blank"
          >
            <img
              alt="placeholder"
              src="[object Object]"
            />
          </a>
        </div>
        <h3
          class="md:text-md flex justify-center text-sm"
        >
          placeholder
        </h3>
        <h4
          class="flex justify-center px-1 text-xs md:text-sm"
        >
          placeholder
        </h4>
      </footer>
    </div>
  </div>
</div>
`;
